src/fides/core/config/redis_settings.py

```diff
@@ -37,17 +37,25 @@ def assemble_connection_url(
             # If the whole URL is provided via the config, preference that
             return v
 
-        db_index = values.get("db_index") if values.get("db_index") is not None else ""
+        db_index = values.get("db_index") or ""
         connection_protocol = "redis"
         params = ""
         use_tls = values.get("ssl", False)
+        user = values.get("user") or ""
+        password = values.get("password") or ""
         if use_tls:
             # If using TLS update the connection URL format
             connection_protocol = "rediss"
             cert_reqs = values.get("ssl_cert_reqs", "none")
             params = f"?ssl_cert_reqs={quote_plus(cert_reqs)}"
 
-        return f"{connection_protocol}://{quote_plus(values.get('user', ''))}:{quote_plus(values.get('password', ''))}@{values.get('host', '')}:{values.get('port', '')}/{db_index}{params}"
+        # Configure a basic auth prefix if either user or password is provided, e.g.
+        # redis://<user>:<password>@<host>
+        auth_prefix = ""
+        if password or user:
+            auth_prefix = f"{quote_plus(user)}:{quote_plus(password)}@"
```

Comment:
Added this to handle the case where someone doesn't have any auth configured on their redis server at all.